packages/local-refresh/actions/ResetDatabase.ts

```diff
@@ -0,0 +1,174 @@
+      },
+      {
+        type: (prev) => (prev === true ? "select" : null),
+        name: "refresh-type",
+        message: "Choose the type of database reset?",
+        choices: [
+          { title: "Full refresh", value: DatabaseRefreshEnum.FULL },
+          { title: "Just migrations", value: DatabaseRefreshEnum.MIGRATIONS },
+        ],
+      },
+    ]);
+
+    if (!response["database-refresh"]) {
+      return false;
+    }
+
+    this.refreshType = response["refresh-type"];
+
+    const etlDirectoryPrompt = await prompts({
+      type: "text",
+      name: "directory",
+      message: "Enter the path to the ETL directory",
+      initial: cache.getKey("etlDirectory") || "../olcs-etl",
+      validate: (value) => (fs.existsSync(value) ? true : "Path does not exist"),
+    });
+
+    cache.setKey("etlDirectory", etlDirectoryPrompt.directory);
+    cache.save();
+
+    this.etlDirectory = etlDirectoryPrompt.directory;
+
+    return this.refreshType !== DatabaseRefreshEnum.NONE;
+  }
+
+  async execute(): Promise<void> {
+    // Full reset requires AWS credentials to pull down the anonymised dataset from S3.
+    if (this.refreshType === DatabaseRefreshEnum.FULL) {
+      if (shell.exec("aws sts get-caller-identity").code !== 0) {
+        console.error(
+          chalk.red(
+            "Error: Valid AWS credentials are required for a full database reset. Authenticate with VOL `nonprod` and retry.",
+          ),
+        );
+
+        return;
+      }
+    }
+
+    const myCnf = dedent`
+    [client]
+    user=root
+    password=olcs
+    host=host.docker.internal
+    port=3306`;
+
+    if (this.refreshType !== DatabaseRefreshEnum.MIGRATIONS) {
+      if (
+        shell.exec(
+          dedent`docker compose exec db /bin/bash -c "\
+          echo '${myCnf}' > ~/.my.cnf; \
+          cd /var/lib/etl \
+          && ./create-base.sh olcs_be
+        "
+      `,
+          {
+            env: {
+              ...process.env,
+              FORCE_COLOR: "1",
+            },
+          },
+        ).code !== 0
+      ) {
+        console.error(chalk.red(`Error: \`create-base.sh\` failed`));
+        return;
+      }
+    }
+
+    if (
+      shell.exec(
+        `docker run \
+        --rm \
+        -e INSTALL_MYSQL=true \
+        -v "$PWD/${this.etlDirectory}/":/liquibase/changelog \
+        -w /liquibase/changelog \
+        liquibase/liquibase \
+          --defaultsFile=liquibase.properties \
+          update \
+          -Ddataset=testdata`,
+        {
+          env: {
+            ...process.env,
+            FORCE_COLOR: "1",
+          },
+        },
+      ).code !== 0
+    ) {
+      console.error(chalk.red(`Error: \`liquibase\` failed`));
+      return;
+    }
+
+    // Fetch file from S3.
+    const latestAnonDatasetCmd = shell.exec(
+      `aws s3 ls s3://devapp-olcs-pri-olcs-deploy-s3/anondata/olcs-db-localdev-anon-prod --recursive 2>/dev/null | sort | tail -n 1 | awk '{print $4}'`,
```

Comment:
Worth moving the bucket name to a const/var as its used more than once in this script?

Reply:
Yeah, good idea! Sorted!